tools/src/main/java/org/apache/kafka/trogdor/coordinator/CoordinatorRestResource.java

```diff
@@ -96,13 +99,25 @@ public Empty destroyTask(@DefaultValue("") @QueryParam("taskId") String taskId)
     }
 
     @GET
-    @Path("/tasks")
-    public TasksResponse tasks(@QueryParam("taskId") List<String> taskId,
+    @Path("/tasks/")
+    public Response tasks(@QueryParam("taskId") List<String> taskId,
             @DefaultValue("0") @QueryParam("firstStartMs") long firstStartMs,
             @DefaultValue("0") @QueryParam("lastStartMs") long lastStartMs,
             @DefaultValue("0") @QueryParam("firstEndMs") long firstEndMs,
-            @DefaultValue("0") @QueryParam("lastEndMs") long lastEndMs) throws Throwable {
-        return coordinator().tasks(new TasksRequest(taskId, firstStartMs, lastStartMs, firstEndMs, lastEndMs));
+            @DefaultValue("0") @QueryParam("lastEndMs") long lastEndMs,
+            @DefaultValue("") @QueryParam("state") String state) throws Throwable {
+        boolean isEmptyState = state.equals("");
+        if (!isEmptyState && !TaskStateType.Constants.values.contains(state)) {
+            return Response.status(400).entity(
+                String.format("State %s is invalid. Must be one of %s",
+                    state, TaskStateType.Constants.values)
```

Comment:
I think you need `Utils#join` here.  Calling `toString` on an array will give you something like `[Values@3343c8b3`

Reply:
`values` is a `List<String>`, so we should be okay.
The reason it is a list is because I wanted to make it unmodifiable. Spotbugs complained that we had a mutable final variable